src/System.Runtime/tests/System/Boolean.cs

```diff
@@ -113,33 +78,53 @@ public static void TestToString()
     [InlineData(true, null, 1)]
     [InlineData(false, false, 0)]
     [InlineData(false, true, -1)]
-    [InlineData(false, null, 0)]
-    public static void TestCompareTo(bool b1, bool b2, int expected)
-    {
-        int i = CompareHelper.NormalizeCompare(b1.CompareTo(b2));
-        Assert.Equal(expected, i);
-    }
-
-    [Theory]
-    [InlineData(true, true, 0)]
-    [InlineData(true, false, 1)]
-    [InlineData(true, null, 1)]
-    [InlineData(false, false, 0)]
-    [InlineData(false, true, -1)]
     [InlineData(false, null, 1)]
-    public static void TestIComparableCompareTo(IComparable b1, IComparable b2, int expected)
+    public static void TestCompareTo(bool b, object obj, int expected)
     {
-        int i = CompareHelper.NormalizeCompare(b1.CompareTo(b2));
-        Assert.Equal(expected, i);
+        if (obj is bool)
+        {
+            Assert.Equal(expected, Math.Sign(b.CompareTo((bool)obj)));
+        }
+
+        IComparable comparable = b;
+        Assert.Equal(expected, Math.Sign(comparable.CompareTo(obj)));
     }
 
     [Theory]
     [InlineData(true, 1)]
     [InlineData(true, "true")]
     [InlineData(false, 0)]
     [InlineData(false, "false")]
-    private static void VerifyCompareToFailures(IComparable b, object obj)
+    private static void TestCompareTo_Invalid(IComparable b, object obj)
     {
-        Assert.Throws<ArgumentException>(null, () => b.CompareTo(obj));
+        Assert.Throws<ArgumentException>(() => b.CompareTo(obj));
```

Comment:
There is no param name, and I don't think its a good idea to include the null param name in the assertion - am I correct?